src/terraform-provider-paperspace/client.go

```diff
@@ -50,93 +55,173 @@ func SetResData(d *schema.ResourceData, m map[string]interface{}, n string) {
+func LogArrayResponse(reqDesc string, reqURL *url.URL, resp *http.Response, body interface{}, err error) {
+	log.Printf("[INFO] Request: %v", reqDesc)
+	log.Printf("[INFO] Request URL: %v", reqURL)
+	log.Printf("[INFO] Response Status: %v", resp.Status)
+	log.Printf("[INFO] Response: %v", resp)
+	log.Printf("[INFO] Response Body: %s", body)
+	log.Printf("[INFO] Error: %v", err)
+}
+
+type ClientConfig struct {
+	APIKey  string
+	APIHost string
 	Region  string
 }
 
 type PaperspaceClient struct {
-	ApiKey      string
-	ApiHost     string
-	Region      string
-	RestyClient *resty.Client
-}
-
-func (c *Config) Client() (PaperspaceClient, error) {
-
-	restyClient := resty.New()
-
-	restyClient.SetDebug(true)
-
-	restyClient.OnBeforeRequest(func(c *resty.Client, req *resty.Request) error {
-		// Explore request object
-		log.Println("[INFO] Request Info:")
-		log.Println("[INFO] client.HostURL", c.HostURL)
-		log.Println("[INFO] client.Header", c.Header)
-		log.Println("[INFO] req.Method", req.Method)
-		log.Println("[INFO] req.URL", req.URL)
-		log.Println("[INFO] req.Body", req.Body)
-		log.Println("[INFO] req.AuthScheme", req.AuthScheme)
-		log.Println("[INFO] req.RawRequest", req.RawRequest)
-		log.Println("[INFO] req.Error", req.Error)
-
-		return nil
-	})
-
-	restyClient.OnAfterResponse(func(c *resty.Client, resp *resty.Response) error {
-
-		// Explore response object
-		log.Println("[INFO] Response Info:")
-		// log.Println("[INFO] Error      :", err)
-		log.Println("[INFO] Status Code:", resp.StatusCode())
-		log.Println("[INFO] Status     :", resp.Status())
-		log.Println("[INFO] Proto      :", resp.Proto())
-		log.Println("[INFO] Time       :", resp.Time())
-		log.Println("[INFO] Received At:", resp.ReceivedAt())
-		log.Println("[INFO] Body       :\n", resp)
-
-		// Explore trace info
-		log.Println("[INFO] Request Trace Info:")
-		ti := resp.Request.TraceInfo()
-		log.Println("[INFO] DNSLookup    :", ti.DNSLookup)
-		log.Println("[INFO] ConnTime     :", ti.ConnTime)
-		log.Println("[INFO] TCPConnTime  :", ti.TCPConnTime)
-		log.Println("[INFO] TLSHandshake :", ti.TLSHandshake)
-		log.Println("[INFO] ServerTime   :", ti.ServerTime)
-		log.Println("[INFO] ResponseTime :", ti.ResponseTime)
-		log.Println("[INFO] TotalTime    :", ti.TotalTime)
-		log.Println("[INFO] IsConnReused :", ti.IsConnReused)
-		log.Println("[INFO] IsConnWasIdle:", ti.IsConnWasIdle)
-		log.Println("[INFO] ConnIdleTime :", ti.ConnIdleTime)
-
-		return nil
-	})
-
-	restyClient.
-		SetHostURL(c.ApiHost).
-		SetHeader("x-api-key", c.ApiKey).
-		SetHeader("Accept", "application/json").
-		SetHeader("Content-Type", "application/json").
-		SetHeader("User-Agent", "terraform-provider-paperspace").
-		SetHeader("ps_client_name", "terraform-provider-paperspace")
+	APIKey     string
+	APIHost    string
+	Region     string
+	HttpClient *http.Client
+}
+
+func (c *ClientConfig) Client() (PaperspaceClient, error) {
+	timeout := 10 * time.Second
+	hc := &http.Client{
```

Comment:
nit: hc, rt could've been called: client and transport

Reply:
go convention is short var names, but i do prefer clear ones myself -- i've changed these